dist/ExtUtils-ParseXS/lib/perlxs.pod

```diff
@@ -1,2065 +1,4803 @@
+        '-=' => \&subtract;
 
-Note that if the version of the PM module is an NV (a floating point
-number), it will be stringified with a possible loss of precision
-(currently chopping to nine decimal places) so that it may not match
-the version of the XS module anymore. Quoting the $VERSION declaration
-to make it a string is recommended if long version numbers are used.
+The rest of the line following the keyword, plus any further lines until
+the next keyword, are interpreted as a space-separated list of overloaded
+operators. There is no check that they are valid operator names. The names
+and symbols will eventually end up within double-quoted strings in the C
+file, so double-quotes need to be escaped; in particular:
 
-=head2 The PROTOTYPES: Keyword
+      OVERLOAD: \"\"
 
-The PROTOTYPES: keyword corresponds to B<xsubpp>'s C<-prototypes> and
-C<-noprototypes> options.  This keyword overrides the command line options.
-Prototypes are disabled by default.  When prototypes are enabled, XSUBs will
-be given Perl prototypes.  This keyword may be used multiple times in an XS
-module to enable and disable prototypes for different parts of the module.
-Note that B<xsubpp> will nag you if you don't explicitly enable or disable
-prototypes, with:
+This could be regarded as an implementation bug, but we're stuck with it
+now.
 
-    Please specify prototyping behavior for Foo.xs (see perlxs manual)
+XSUBs used for overload methods are invoked with the same arguments as
+Perl subroutines would be: for example, an overloaded binary operator will
+trigger a call to the XSUB method with the first argument being an
+overloaded object representing one of the two operands of the binary
+operator; the second being the other operand (which may or may not be an
+object); and third, a swap flag. See L<overload> for the full details
+of how these functions will be called, with what arguments. Note that
+C<swap> can in fact be undef in addition to false, to indicate an assign
+overload such as C<+=>. If this difference is important to your code, then
+declare C<swap> as type C<SV*> so that you can use SvOK() and SvTRUE()
+on it.
 
-To enable prototypes:
+Bitwise operator methods sometimes take extra arguments: in
+particular under C<use feature 'bitwise'>. So you may want to use an
+ellipsis (something like C<(lobj, robj, swap, ...)>) to skip them.
 
-    PROTOTYPES: ENABLE
+The net effect of the C<OVERLOAD> keyword is to add some extra code to
+the boot XSUB to register this XSUB as the handler for the specified
+overload actions, in the same way that C<use overload> does for Perl
+methods.
 
-To disable prototypes:
+See also the file-scoped L<FALLBACK|/The FALLBACK: Keyword> keyword for
+details of how to set the fallback behaviour for the current package.
 
-    PROTOTYPES: DISABLE
+Note that C<OVERLOAD> shouldn't be mixed with the L<ALIAS|/The ALIAS:
+Keyword> keyword; the value of C<ix> will be undefined for any overload
+method call.
```

Comment:
Aside: Is this an implementation bug that we should consider fixing?

Reply:
it's on my list of things to address.